src/config/ts-node.ts

```diff
@@ -1,5 +1,6 @@
 import * as fs from 'fs'
 import * as path from 'path'
+import * as TSNode from 'ts-node'
```

Comment:
`ts-node` should be moved to the `dependencies` object:
https://github.com/oclif/core/blob/main/package.json#L76

Reply:
I can move it but I'm not sure that it's necessary since we're only importing it for a type. 

If you compile and look at this file there's no `ts-node` import,
```javascript
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tsPath = void 0;
const fs = require("fs");
const path = require("path");
const settings_1 = require("../settings");
const util_1 = require("../util");
const util_2 = require("./util");
// eslint-disable-next-line new-cap
const debug = (0, util_2.Debug)('ts-node');
```

Reply:
this `require.resolve` will try to look up the location of `ts-node` at runtime.

compiled output:
```js
function registerTSNode(root) {
    const tsconfig = loadTSConfig(root);
    if (!tsconfig)
        return;
    debug('registering ts-node at', root);
    const tsNodePath = require.resolve('ts-node', { paths: [root, __dirname] });
...
```

https://github.com/oclif/core/pull/650/files/83c60b3b64534ccae2a3c3251d3b492f5fc383e4#diff-5622386b4043e10ee71bec48dcb5cefd6c9cae20cecc1124bd40d82ac82ab8b6R45